Kcho/update path detection to use regrex

kcho opened the pull request:
I have noticed that `lochness.XNAT` module will download data to a subdirectory (eg- 2001_MR1) under `mri` directory in the phoenix structure. It could also be the case that other teams may add extra subdirectories under `box` and `mediaflux` as well. To account for this extra level of directories under the datatype directory, I've updated `FileInPhoenixBIDS` and `FileInPhoenix` classes to use regrex to be more flexible under these situations. Tested using `pytest test/dpanonymize/test_dpanonymize.py`.

Thanks.

kcho committed:
feat: eeg template added

kcho committed:
feat: eeg template added

kcho committed:
Merge branch 'kcho/prescient' of https://github.com/AMP-SCZ/lockness into kcho/prescient

kcho committed:
fix: kwargs error fix

kcho committed:
feat: conflict resolve for EEG

kcho committed:
fix: conflix resolve

kcho committed:
feat: simplify dtype and other path detections using regrex instead of absolute path patterns

tashrifbillah commented on dpanonymize/__init__.py:
```diff
@@ -53,18 +67,37 @@ def anonymize(self, **kwargs) -> None:
 
     def __repr__(self):
-        return f"<{self.file_path.name}>"
+        return f"<{self.file_path.name} {self.dtype}"
```
Are you missing ending `>`?

kcho replied:
5846515

tashrifbillah commented on dpanonymize/__init__.py:
```diff
@@ -53,18 +67,37 @@ def anonymize(self, **kwargs) -> None:
 class FileInPhoenix(FileInPhoenixBIDS):
-    '''NON-BIDS PHOENIX file class used to grab file information'''
-    def __init__(self, file_path):
+    '''NON-BIDS PHOENIX file class used to grab file information
```
If `NON-BIDS`, why does the argument say `FileInPhoenixBIDS`?

kcho replied:
It's just inheriting from `FileInPhoenixBIDS` class. `FileInPhoenix` is for non-bids. Thanks for double checking.

tashrifbillah commented on dpanonymize/__init__.py:
```diff
@@ -140,6 +175,8 @@ def lock_lochness(Lochness: 'Lochness',
             })
         pii_table_loc = phoenix_root.parent / 'pii_convert.csv'
         df.to_csv(pii_table_loc)
+        kwargs['pii_table_loc'] =  pii_table_loc
```
This and the following line already exists in PR #14 which should result in conflict. This two PRs should be in tandem: 14-->15-->master

kcho replied:
Good to know that we can do it this way. Let me try changing the base to #14 -> #15

kcho replied:
Just changed the #14's base to #15, and merged it. Thanks for letting me know this.

tashrifbillah commented on dpanonymize/__init__.py:
```diff
@@ -77,9 +110,11 @@ def get_file_objects_from_phoenix(root_dir: Union[Path, str],
+
     for root, _, files in os.walk(root_dir):
         for file in files:
-            full_path = Path(root) / file
```
What was incomplete before that we have to insert `str( )`?

kcho replied:
The `FileInPhoenixBIDS` and `FileInPhoenix` classes require string inputs now, since we're using regrex to catch names. So I'm making the Path objects into strings before putting into the classes.

kcho committed:
typo: add ">"

kcho committed:
Merge pull request #14 from AMP-SCZ/kcho/prescient

EEG module has been added

tashrifbillah commented on dpanonymize/__init__.py:
```diff
@@ -17,17 +18,30 @@
+
+    Key Arguments:
+        - file_path: file path under BIDS phoenix structure, str
+          Example file_path
```
I feel like this slide should be attached to every sentence related to BIDS:

https://github.com/AMP-SCZ/lochness/pull/40#discussion_r671856493

kcho replied:
Yeap you've pointed out that we should do this. I'll add these doc strings to places when I work on another branch of cleaning things up, after we have the fully working pipeline. Thanks.

tashrifbillah commented:
#### Clarifiying discussion

> It could also be the case that other teams may add extra subdirectories under box and mediaflux as well. To account for this extra level of directories under the datatype directory

In regard to your description, I am trying to understand the need for this new code:

https://github.com/AMP-SCZ/dpanonymize/blob/fa99674544b375a1aa3145be56ade86bcc6986d5/dpanonymize/__init__.py#L29-L45

The above code correctly detects processed, study, subject, and data groups for both beginning and ending extra paths:

```python
file_path="/Some/Path/EXTRA/PROTECTED/BWH/raw/subject01/actigraphy/file"
In [98]: search.group('processed')
Out[98]: 'raw'

In [99]: search.group('study')
Out[99]: 'BWH'

In [100]: search.group('dtype')
Out[100]: 'actigraphy'

In [101]: search.group('subject')
Out[101]: 'subject01'
```

```python
file_path="/Some/Path/to/PROTECTED/BWH/raw/subject01/actigraphy/EXTRA/file"
In [105]: search.group('subject')
Out[105]: 'subject01'

In [106]: search.group('dtype')
Out[106]: 'actigraphy'

In [107]: search.group('study')
Out[107]: 'BWH'

In [108]: search.group('processed')
Out[108]: 'raw'
```

Did you observe the same? If yes, the description should be updated as:

> It could also be the case that other teams may add extra subdirectories under **box/mediaflux and datatype** directories. To account for this extra level of **beginning and ending** directories ...

kcho committed:
doc: FileInPhoenix doc string for being able to detect files under extra folder levels under the datatype root dir

kcho committed:
Merge branch 'kcho/update_path_detection_to_use_regrex' of https://github.com/AMP-SCZ/lockness into kcho/update_path_detection_to_use_regrex

kcho commented:
> #### Clarifiying discussion
> > It could also be the case that other teams may add extra subdirectories under box and mediaflux as well. To account for this extra level of directories under the datatype directory
> 
> In regard to your description, I am trying to understand the need for this new code:
> 
> https://github.com/AMP-SCZ/dpanonymize/blob/fa99674544b375a1aa3145be56ade86bcc6986d5/dpanonymize/__init__.py#L29-L45
> 
> The above code correctly detects processed, study, subject, and data groups for both beginning and ending extra paths:
> 
> ```python
> file_path="/Some/Path/EXTRA/PROTECTED/BWH/raw/subject01/actigraphy/file"
> In [98]: search.group('processed')
> Out[98]: 'raw'
> 
> In [99]: search.group('study')
> Out[99]: 'BWH'
> 
> In [100]: search.group('dtype')
> Out[100]: 'actigraphy'
> 
> In [101]: search.group('subject')
> Out[101]: 'subject01'
> ```
> 
> ```python
> file_path="/Some/Path/to/PROTECTED/BWH/raw/subject01/actigraphy/EXTRA/file"
> In [105]: search.group('subject')
> Out[105]: 'subject01'
> 
> In [106]: search.group('dtype')
> Out[106]: 'actigraphy'
> 
> In [107]: search.group('study')
> Out[107]: 'BWH'
> 
> In [108]: search.group('processed')
> Out[108]: 'raw'
> ```
> 
> Did you observe the same? If yes, the description should be updated as:
> 
> > It could also be the case that other teams may add extra subdirectories under **box/mediaflux and datatype** directories. To account for this extra level of **beginning and ending** directories ...

It's to support extra folders under the datatype root directory, which is already the case for the data pulled by XNAT.
eg) `lochness.XNAT` puts another folder under `mri` , eg) `raw/subject01/mri/2001_MR`.

Doc string has been updated to add a little more elaboration b31a9f3

tashrifbillah committed:
minor wording

https://github.com/AMP-SCZ/dpanonymize/commit/b31a9f323b8636b61f90607bc78d11eae3a2fe89#r56472472

tashrifbillah reviewed (changes requested):
Please consider merging my suggestion #16 and we are good to go. Thank you.

kcho committed:
Merge pull request #16 from AMP-SCZ/patch-1

minor wording

kcho commented:
> Please consider merging my suggestion #16 and we are good to go. Thank you.

Merged. Thanks
